ui/app/src/components/GlobalConfigManagement/GlobalConfigManagement.tsx

```diff
@@ -86,14 +76,52 @@ export function GlobalConfigManagement() {
     ];
 
     forkJoin(requests).subscribe(([sample, content]) => {
+      const sessionContent = sessionStorage.getItem(sessionStorageKey);
+
       setLastSavedContent(content);
-      setContent(content);
-      setContentSize(content.length);
+      if (sessionContent) {
+        setContent(sessionContent);
+        setContentSize(sessionContent.length);
+        setHasChanges(true);
+        sessionStorage.removeItem(sessionStorageKey);
+      } else {
+        setContent(content);
+        setContentSize(content.length);
+      }
       setSample(sample);
       setEnable(false);
     });
   });
 
+  useUnmount(() => {
+    if (hasChangesRef.current) {
+      sessionStorage.setItem(sessionStorageKey, aceEditorRef.current.getValue());
+      const eventId = 'unsavedConfigManagementChangesConfirmation';
+      dispatch(
+        showConfirmDialog({
+          body: <FormattedMessage defaultMessage="You left unsaved changes. Go back and continue editing?" />,
+          onCancel: batchActions([closeConfirmDialog(), dispatchDOMEvent({ id: eventId, button: 'cancel' })]),
+          onOk: batchActions([closeConfirmDialog(), dispatchDOMEvent({ id: eventId, button: 'ok' })]),
+          okButtonText: <FormattedMessage defaultMessage="Continue editing" />,
+          cancelButtonText: <FormattedMessage defaultMessage="Discard changes" />
+        })
+      );
+      createCustomDocumentEventListener<{ button: 'ok' | 'cancel' }>(eventId, ({ button }) => {
+        if (button === 'ok') {
+          navigate('/global-config');
+        } else {
+          sessionStorage.removeItem(sessionStorageKey);
+        }
+      });
+    }
+  });
+
+  useBeforeUnload((event) => {
+    if (hasChangesRef.current) {
+      event.preventDefault();
+    }
+  });
```

Comment:
Can you specify any kind of message to display? How does this work?

Reply:
Modern browsers do not support custom messages in the 'beforeunload' popup. For example see https://chromestatus.com/feature/5349061406228480. The message that shows is:
```
Reload/Leave Site? (depending on action)
Changes you made may not be saved.
```